frontend/src/components/Search/Parser/listener.ts

```diff
@@ -110,10 +115,13 @@ export class SearchListener extends SearchGrammarListener {
 	}
 
 	exitKey_val_search_expr = (_ctx: Key_val_search_exprContext) => {
-		this.currentExpression.value = this.currentExpression.text.substring(
-			this.currentExpression.key.length +
-				this.currentExpression.operator.length,
+		// everything after the key, operator, and whitespace
+		const regex = new RegExp(
+			`^${this.currentExpression.key}\\s*${this.currentExpression.operator}\\s*(.*)$`,
```

Comment:
not sure what the culprit is, but there seems to be a new flicker in the dropdown after a value is entered
https://www.loom.com/share/d5ef948087244c18b0da8292b6792ede

Reply:
See https://github.com/highlight/highlight/pull/9681#issuecomment-2532068336 - believe this is resolved!